### ISSUE TITLE
[Dark Mode] The color contrast for selected DataGridViewTextBox is insufficient

### ISSUE THREAD

**Issue by Amy-Li03** [dotnet/winforms#13780]
### .NET version

.NET 10 SDK build: 10.0.100-rc.1.25401.103

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Occurs when DarkMode feature involved

### Issue description

When the application is in Dark Mode, the color contrast of DataGridViewTextBox is insufficient.
This issue does not occur in Classic Mode.

<img width="1158" height="740" alt="Image" src="https://github.com/user-attachments/assets/088975a6-6677-4865-bdfc-ba4da0ee83d2" />

### Steps to reproduce

1. Create Winforms .NET application with targeting .NET 10
2. Add a DataGridView control with DataGridViewTextBoxColumn in form designer
3. Set app under Dark mode in program.cs file:
```
#pragma warning disable WFO5001
      Application.SetColorMode(SystemColorMode.Dark);
#pragma warning restore WFO5001
```
4. Build and run project
5. Input something in DataGridViewTextBox, then select this cell
6. Observe the color contrast between selected DataGridViewTextBox's Text color and Back color
